File: playground/bubble_sort.py




def bubble_sort(array):
    i = 0
    j = i
    while i < len(array):
        j = i
        while j < len(array):
            if array[j] < array[j-1]:
                temp = array[j]
                array[j] = array[j-1]
                array[j-1] = temp
            j+=1        
        i +=1

a = [12, 200, 1, 3000, 30023, 123, 12]
bubble_sort(a)
print(a)

File: playground/bubble_sort.js


function bubble_sort(array) {

  let i = 0;
  let j = 0;

  while (i < array.length) {
      j = i;
      while (j < array.length) {
        if (array[j] < array[j-1]) {
          let temp = array[j];
          array[j] = array[j-1];
          array[j-1] = temp;
        }
        j +=1;
      }
    i +=1;
  }
  return array;
}


